src/index.ts

```diff
@@ -28,31 +30,46 @@ function idToAsset(id: string): AssetID {
   // TODO use caip package to do this once it supports assetIds
   const [chainid, assetType, tokenId] = id.split('_');
   const [namespace, reference] = assetType.split('.');
-
-  if (!['erc721', 'erc1155'].includes(namespace)) 
-    throw new Error('Not a valid NFT namespace');
   
   return {
-    chainId: ChainID.parse(chainid.replace('.', ':')),
+    chainId: new ChainID(ChainID.parse(chainid.replace('.', ':'))),
     namespace,
     reference,
     tokenId
   }
 }
 
-async function assetToAccount(asset: AssetID, timestamp: number, customSubgraph?: SubGraphUrls): Promise<AccountID[]> {
+async function assetToAccount(
+  asset: AssetID, 
+  timestamp: number, 
+  customSubgraph?: SubGraphUrls
+): Promise<AccountID[]> {
   // we want to query what block is at the timestamp IFF it is an (older) existing timestamp
   let queryBlock: number = undefined;
   if (timestamp && !isWithinLastBlock(timestamp)) {
     queryBlock = await blockAtTime(timestamp);
   }
 
   let owners: string[];
-  if (asset.namespace === ErcNamespace.ERC721) {
-    owners = [ await erc721OwnerOf(asset, queryBlock, customSubgraph?.erc721) ];
+  // dealing with eth mainnet
+  if (asset.chainId.toString() === ETH_CAIP2_CHAINID) {
+    let ethSubgraphUrls = undefined;
+
+    if (customSubgraph && customSubgraph[ETH_CAIP2_CHAINID]) {
+      ethSubgraphUrls = customSubgraph[ETH_CAIP2_CHAINID];
+    }
+
+    if (asset.namespace === 'erc721') {
+      owners = [ await erc721OwnerOf(asset, queryBlock, ethSubgraphUrls?.erc721) ];
+    } else if (asset.namespace === 'erc1155') {
+      owners = await erc1155OwnersOf(asset, queryBlock, ethSubgraphUrls?.erc1155);
+    } else {
+      throw new Error(`Unrecognized ERC Namespace: ${asset.namespace}`);
+    }
   } else {
-    owners = await erc1155OwnersOf(asset, queryBlock, customSubgraph?.erc1155);
+    throw new Error('Caip2 chains besides ETH are not supported yet')
```

Comment:
Would love your thoughts on what to do here!

Reply:
This seems fine. Although it would be nice to not throw an error if someone passes a compatible subgraph for the given chain, i.e. erc721s on xDAI chain.

Reply:
Sounds good - I'll just remove the eth caip2 if-else wrapping and only restrict it by asset namespace.